src/fifolock.jl

```diff
@@ -0,0 +1,150 @@
+@static if VERSION >= v"1.10"
```

Comment:
Can you leave a comment on this indicating why it is v1.10 only?

Reply:
Uh. Mostly because I didn't want to test/debug it on older versions. Also, I think CI runs Julia 1.6, but per-field atomics were added only in Julia 1.7.

Reply:
Sorry, i meant a _code comment,_ not a GH comment. The code should indicate why this is 1.10 only.
If the answer is "We haven't tested this on older versions yet," then that's fine, and some future contributor could pull it back to older julias too. But only if they have clarity on why this is there.